packages/react-dom/src/__tests__/ReactFlightIntegration-test.js

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+// Polyfills for test environment
+global.ReadableStream = require('@mattiasbuelens/web-streams-polyfill/ponyfill/es6').ReadableStream;
+global.TextDecoder = require('util').TextDecoder;
+
+let Scheduler;
+let Stream;
+let React;
+let ReactDOM;
+let ReactFlightDOMServer;
+let ReactFlightDOMClient;
+
+describe('ReactFlightIntegration', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    Scheduler = require('scheduler');
+    Stream = require('stream');
+    React = require('react');
+    ReactDOM = require('react-dom');
+    ReactFlightDOMServer = require('react-dom/unstable-flight-server');
+    ReactFlightDOMClient = require('react-dom/unstable-flight-client');
+  });
+
+  function getTestStream() {
+    let writable = new Stream.PassThrough();
+    let readable = new ReadableStream({
+      start(controller) {
+        writable.on('data', chunk => {
+          controller.enqueue(chunk);
+        });
+        writable.on('end', () => {
+          controller.close();
+        });
+      },
+    });
+    return {
+      writable,
+      readable,
+    };
+  }
+
+  async function waitForDOMChange(container) {
```

Comment:
This is kinda wonky but I wanted to check for next mutation specifically rather than manually wait for models. Since React itself would wait for models and I want to test it waits correctly.

Reply:
Isn’t this what async act is for? Why isn’t it sufficient? It’s supposed to force flush even if it suspends and wait for nested promises. 

What’s missing and can we fix that?

Reply:
Lol I forgot about it. Let me try it